src/expr/scalar/func.rs

```diff
@@ -159,12 +164,11 @@ fn cast_int64_to_bool<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(a.unwrap_int64() != 0)
 }
 
-fn cast_int64_to_int32<'a>(a: Datum<'a>) -> Datum<'a> {
-    // TODO(benesch): we need to do something better than panicking if the
-    // datum doesn't fit in an int32, but what? Poison the whole dataflow?
-    // The SQL standard says this an error, but runtime errors are complicated
-    // in a streaming setting.
-    Datum::from(i32::try_from(a.unwrap_int64()).unwrap())
+fn cast_int64_to_int32<'a>(a: Datum<'a>) -> Result<Datum<'a>, failure::Error> {
+    match i32::try_from(a.unwrap_int64()) {
+        Ok(n) => Ok(Datum::from(n)),
+        Err(_) => Err(format_err!("integer out of range")),
```

Comment:
Should this be `bail!()` to match other examples in the code? No opinion myself, especially if there is a reason, but consistency is great if possible! :D

Reply:
I personally use `bail!` only for early returns, and `format_err!` for non-early returns—but I can see how that might be more confusing for other folks! I'll change if I go back through CI on this PR. Otherwise will punt to the follow-on PR to ease the rebase.